rdr_service/dao/sample_summary_dao.py

```diff
@@ -0,0 +1,94 @@
+from datetime import datetime
+from typing import List
+
+from sqlalchemy.orm import Session
+
+from rdr_service.model.biobank_stored_sample import BiobankStoredSample
+from rdr_service.model.sample_order_status import SampleOrderStatus
+from rdr_service.model.sample_receipt_status import SampleReceiptStatus
+from rdr_service.participant_enums import OrderStatus, SampleStatus
+
+
+class SampleSummaryDao:
+    @classmethod
+    def upsert_order_data(cls, participant_id: int, status: OrderStatus, test: str, time: datetime, session: Session):
+        order_data = session.query(SampleOrderStatus).filter(
+            SampleOrderStatus.participant_id == participant_id,
+            SampleOrderStatus.test_code == test
+        ).one_or_none()
+
+        if order_data:
+            order_data.status = status
+            order_data.status_time = time
+        else:
+            session.add(
+                SampleOrderStatus(
+                    participant_id=participant_id,
+                    test_code=test,
+                    status=status,
+                    status_time=time
+                )
+            )
+
+    @classmethod
+    def refresh_receipt_data(cls, participant_id: int, new_sample: BiobankStoredSample, session: Session):
+        receipt_data = session.query(SampleReceiptStatus).filter(
+            SampleReceiptStatus.participant_id == participant_id,
+            SampleReceiptStatus.test_code == new_sample.test
+        ).one_or_none()
+
+        if receipt_data:
+            sample_list = session.query(BiobankStoredSample).filter(
+                BiobankStoredSample.biobankId == new_sample.biobankId,
+                BiobankStoredSample.test == new_sample.test
+            ).all()
+            sample_list = [*sample_list, new_sample]
+
+            status_sample = None
+            non_disposed_samples = [
+                sample for sample in sample_list if sample.status < SampleStatus.SAMPLE_NOT_RECEIVED
```

Comment:
Neat way to filter for the good and bad samples, but if more good sampleStatus Enums are added this may break later. It may be better to put the good and bad sample status enums in separate lists?

Reply:
Yeah, it'd be good to deal with this kind of thing before it becomes a problem. But for this work I'm just looking to replicate the existing logic in the SQL that processes the statuses (just to reduce the chances of unintended changes/bugs 😅).